cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -299,6 +317,104 @@ pub fn chain_to_prefix_operations(
+}
+
+pub fn authorized_address_operations(
+    deps: DepsMut,
+    sender: Addr,
+    operation: Vec<AuthorizedAddressInput>,
+) -> Result<Response, ContractError> {
+    let response = Response::new();
+    for operation in operation {
+        let addr = operation.addr;
+        let source_chain = operation.source_chain.to_lowercase();
+
+        // Only contract governor can call by-chain permission CRUD operations
+        check_is_contract_governor(deps.as_ref(), sender.clone())?;
+
+        match operation.operation {
+            Operation::Set => {
+                let authorized_addr = AUTHORIZED_ADDRESSES
+                    .may_load(deps.storage, &source_chain)
+                    .unwrap_or_default();
+
+                if authorized_addr.eq(&Some(addr.clone())) {
+                    return Err(ContractError::CustomError {
+                        msg: format!(
+                            "{} is already an authorized address for source chain {}",
+                            addr, source_chain
+                        ),
+                    });
+                }
+
+                AUTHORIZED_ADDRESSES.save(deps.storage, &source_chain, &addr)?;
+                response.clone().add_attribute(
+                    "set_authorized_address",
+                    format!("{}-{}", source_chain, addr),
+                );
+            }
+            Operation::Change => {
+                let authorized_addr = AUTHORIZED_ADDRESSES
+                    .load(deps.storage, &source_chain)
+                    .map_err(|_| ContractError::CustomError {
+                        msg: format!(
+                            "No authorized address found for source chain {}",
+                            source_chain
+                        ),
+                    })?;
+
+                if !authorized_addr.eq(&addr) {
+                    return Err(ContractError::CustomError {
+                        msg: format!(
+                            "{} is not currently an authorized address for source chain {}",
+                            addr, source_chain
+                        ),
+                    });
+                }
+
+                AUTHORIZED_ADDRESSES.remove(deps.storage, &source_chain);
+                AUTHORIZED_ADDRESSES.save(deps.storage, &source_chain, &addr)?;
+                response.clone().add_attribute(
+                    "change_authorized_address",
+                    format!("{}-{}", source_chain, addr),
+                );
+            }
+            Operation::Remove => {
+                let authorized_addr = AUTHORIZED_ADDRESSES
+                    .load(deps.storage, &source_chain)
+                    .map_err(|_| ContractError::CustomError {
+                        msg: format!(
+                            "No authorized address found for source chain {}",
+                            source_chain
+                        ),
+                    })?;
+
+                if !authorized_addr.eq(&addr) {
```

Comment:
Can we move these checks outside the match? IIUC the code is exactly the same on all cases here